headache.php

```diff
@@ -283,3 +283,29 @@ function disable_indexing()
 }
 
 add_action('pre_option_blog_public', __NAMESPACE__ . '\\disable_indexing');
+
+// Strip all HTML tags except those we have whitelisted
+function remove_excessive_html_tags($in): array
+{
+    $in['paste_preprocess'] = "function(plugin, args) {
+        // Strip all HTML tags except those we have whitelisted
+        var whitelist = 'h1,h2,h3,h4,h5,h6,p,ol,ul,a';
+        var stripped = jQuery('<div>' + args.content + '</div>');
+        var els = stripped.find('*').not(whitelist);
+
+        for (var i = els.length - 1; i >= 0; i--) {
+            var e = els[i];
+            jQuery(e).replaceWith(e.innerHTML);
+        }
+
+        // Strip all class and id attributes
+        stripped.find('*').removeAttr('id').removeAttr('class');
+
+        // Return the clean HTML
+        args.content = stripped.html();
+    }";
+
+    return $in;
+}
+
+add_filter('tiny_mce_before_init', remove_excessive_html_tags);
```

Comment:
```suggestion
// Sanitize HTML content when pasting in TinyMCE editor.
function sanitize_tiny_mce_html_content(array $config): array
{
    $config['paste_preprocess'] = "function(plugin, args) {
        // Allow specific HTML tags while sanitizing the content
        const allowedTags = new Set(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p', 'ol', 'ul', 'a']);
        const sanitizedContent = document.createElement('div');
        sanitizedContent.innerHTML = args.content;

        // Remove elements not in the allowed tags
        sanitizedContent.querySelectorAll('*').forEach(element => {
            if (!allowedTags.has(element.tagName.toLowerCase())) {
                element.replaceWith(...element.childNodes); // Replace with child nodes
            }
        });

        // Strip class and id attributes
        sanitizedContent.querySelectorAll('*').forEach(element => {
            element.removeAttribute('id');
            element.removeAttribute('class');
        });

        // Return the clean HTML
        args.content = sanitizedContent.innerHTML;
    }";

    return $config;
}

add_filter('tiny_mce_before_init', __NAMESPACE__ . '\\sanitize_tiny_mce_html_content');
```

Reply:
@puredazzle I've updated the function name and removed jQuery. Please test and see if this works.